sdk/python/kfp/dsl/_ops_group.py

```diff
@@ -217,10 +217,11 @@ class ParallelFor(OpsGroup):
     def __init__(self,
                  loop_args: Union[_for_loop.ItemList,
                                   _pipeline_param.PipelineParam],
-                 parallelism: int = None):
-        if parallelism and parallelism < 1:
+                 parallelism: Optional[int] = None):
+        parallelism = parallelism or 0
+        if parallelism < 0:
             raise ValueError(
-                'ParallelFor parallism set to < 1, allowed values are > 0')
+                f'ParallelFor parallelism must be >= 0. Got: {parallelism}.')
```

Comment:
Remind me: parallelism being 0 meaning unlimited parallelism, right? Just make sure we have this documented in docstring for your PR targets v2 (master branch).

Reply:
Yes, 0 means "unconstrained".
I'll update the v2 PR to include that in the docstring